### ISSUE TITLE
TabBar DataProvider doesn't work

### ISSUE THREAD

**Issue by pashminakazi** [apache/royale-asjs#366]
I compile my own example and tourdeflex TabBarExample
it shows tabBar but doesn't set dataProvider
please implement TabBar Dataprovider Functionality

**Comment by aharui** [comment]
Please provide a simple example of how you want it to work.  Implementing the solution yourself would be a great opportunity for you to learn more about writing Royale framework code.  It is in your employer's best interests to have you learn this in order to better ensure that framework bugs don't become major blocking issues for you.

**Comment by aharui** [comment]
I see you are using Spark TabBar.  I don't think anyone's used that before.  I'll look into it tomorrow.  I assume you are getting an exception in the console?

**Comment by carlosrovira** [comment]
Hi just something maybe you should know. I recently remove "id" dependency in Jewel TabBar. Since HTML use of id is different from Flash, I recommend to use other strategy. In Jewel I use "name", I think is better to avoid the "id/localId" problems.
HTH

**Comment by piotrzarzycki21** [comment]
> Hi just something maybe you should know. I recently remove "id" dependency in Jewel TabBar. Since HTML use of id is different from Flash, I recommend to use other strategy. In Jewel I use "name", I think is better to avoid the "id/localId" problems.
> HTH

Why avoid `localId`? In your case you were changing API of some component and go with `name `property, but from developer perspective `localId ` is the way to go.

**Comment by carlosrovira** [comment]
Don't think so. I think "id" was perfect in flash since ids are local. Our current "localId" seems more that a temporal solution taking into account the discussion weeks ago. Right now is working, but I expect it to evolve to something more final. In the mean time, check for "name" that is part of the API interface Jewel containers that do that kind of navigation expect, seems more correct to me.

Anyway, as I state, localId is temporal, so if we finaly get "id" working as localId in Royale, it would be ok for me to change to support that. So we should think in the actual solution could be temporal depending on Royale Compiler solving this issues

**Comment by piotrzarzycki21** [comment]
> Anyway, as I state, localId is temporal, so if we finaly get "id" working as localId in Royale, it would be ok for me to change to support that. So we should think in the actual solution could be temporal depending on Royale Compiler solving this issues

Could you point me that discussion or give me summary ?

**Comment by carlosrovira** [comment]
yeah, Search for topic **Revisiting the old debate: 'localId vs. id'**, started by Greg. He wanted to work on that, but current work in our real App was making him to defer it.

**Comment by piotrzarzycki21** [comment]
@carlosrovira got it! Thank you! :)
